### PR TITLE
[extension/basicauth] Accept empty usernames ; remove extra validation code

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Drop the validation that username is a required non-empty string.
Remove extraneous validation logic, use config.Validate

### CODE REVIEW

**Review by frzifus — APPROVED** [review]

File: extension/basicauthextension/extension.go [hunk 1]
```diff
@@ -33,18 +33,14 @@ type basicAuth struct {
 	matchFunc  func(username, password string) bool
 }
 
-func newClientAuthExtension(cfg *Config) (auth.Client, error) {
-	if cfg.ClientAuth == nil || cfg.ClientAuth.Username == "" {
-		return nil, errNoCredentialSource
-	}
-
```
Comment (dmitryax) [inline extension/basicauthextension/extension.go:40]: Looks like Config.Validate misses the `ClientAuth.Username == ""` check

File: .chloggen/basicauth_validation.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+subtext:
+
+# If your change doesn't affect end users or the exported elements of any package,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: []
```
Comment (dmitryax) [inline .chloggen/basicauth_validation.yaml:27]: I think it's not needed for the end used

File: extension/basicauthextension/extension.go (hunk 1 above)
Comment (atoulme) [reply extension/basicauthextension/extension.go:40]: That check is actually invalid according to RFC 2617. I was a bit confused so I looked it up:
https://datatracker.ietf.org/doc/html/rfc2617#section-2
As described in this SO answer:
https://stackoverflow.com/a/74658311/407903

File: .chloggen/basicauth_validation.yaml (hunk 2 above)
Comment (atoulme) [reply .chloggen/basicauth_validation.yaml:27]: given the discussion above, it might be needed, I'll change the note.

**Review by dmitryax — APPROVED** [review]

**Comment by atoulme** [comment]
@dmitryax ready to merge?

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**